workflow/artifactrepositories/artifactrepositories.go

```diff
@@ -52,7 +53,7 @@ func (s *artifactRepositories) Resolve(ctx context.Context, ref *wfv1.ArtifactRe
 	}
 	for _, r := range refs {
 		resolvedRef, _, err := s.get(ctx, r)
-		if apierr.IsNotFound(err) {
+		if err != nil && (apierr.IsNotFound(err) || strings.Contains(err.Error(), "config map missing key")) {
```

Comment:
Move this string to a constant so it can be reused by `get()` when emitting this error?